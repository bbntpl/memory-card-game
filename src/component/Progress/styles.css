.progress-bar{
	position: relative;
	width: min(100%, 300px);
	height: 30px;
	border: 1px solid black;
	justify-self: center;
	border-radius: 5px;
}

.progress-bar__background {
	background: purple;
	z-index: 30;
	border-radius: 5px;
}

.progress-bar__background, .progress-bar__overlay {
	top: 0;
	left: 0;
	height: 100%;
}

.progress-bar__overlay {
	position: absolute;
	width: 100%;
	background-color: rgba(0, 0, 0, .4);
	color: aliceblue;
	text-align: center;
	padding: 6px 0;
	font-size: 12px;
	z-index: 50;
	font-weight: 500;
}

@media screen and (max-width: 360px) {
	.progress-bar, .progress-bar__background {
		border-radius: 0;
		border: none;
	}
}