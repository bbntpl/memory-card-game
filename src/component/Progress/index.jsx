import './styles.css';

export default function Progress(props) {
	const { totalUniqueVisits, totalRounds } = props;
	const { current } = totalUniqueVisits;
	const roundedUniqueVisitsPercentage =
		`${Math.floor((current / totalRounds) * 100)}%`
	return (
		<div className='progress-bar'>
			<div
				className='progress-bar__background'
				style={{ width: roundedUniqueVisitsPercentage }}
			>
			</div>
			<div className='progress-bar__overlay'>
				{`${roundedUniqueVisitsPercentage}(
					${current}/${totalRounds})`}
			</div>
		</div>
	)
}