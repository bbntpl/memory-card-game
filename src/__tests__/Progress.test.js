import { create, act} from 'react-test-renderer';
import Progress from '../component/Progress';

it('progress bar width must be the calculation from two variables', () => {
	const component = create(
		<Progress
			totalUniqueVisits={{ current:4 }}
			totalRounds={40}
		/>
	);
	expect(component.toJSON()).toMatchSnapshot();

	act(() => {
		component.update(
			<Progress
				totalUniqueVisits={{ current: 20 }}
				totalRounds={40}
			/>
		)
	});

	expect(component.toJSON()).toMatchSnapshot();

	act(() => {
		component.update(
			<Progress
				totalUniqueVisits={{ current: 40 }}
				totalRounds={40}
			/>
		)
	});

	expect(component.toJSON()).toMatchSnapshot();
})