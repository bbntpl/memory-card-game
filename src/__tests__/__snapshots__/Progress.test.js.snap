// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress bar width must be the calculation from two variables 1`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar__background"
    style={
      Object {
        "width": "10%",
      }
    }
  />
  <div
    className="progress-bar__overlay"
  >
    10%(
					4/40)
  </div>
</div>
`;

exports[`progress bar width must be the calculation from two variables 2`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar__background"
    style={
      Object {
        "width": "50%",
      }
    }
  />
  <div
    className="progress-bar__overlay"
  >
    50%(
					20/40)
  </div>
</div>
`;

exports[`progress bar width must be the calculation from two variables 3`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar__background"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="progress-bar__overlay"
  >
    100%(
					40/40)
  </div>
</div>
`;
